packages/core/src/browser/shell.ts

```diff
@@ -64,7 +62,7 @@ export interface DockLayoutData extends DockPanel.ILayoutConfig {
     activeWidgets?: Widget[]
 }
 
-export const MAINAREA_TABBAR_CONTEXT_MENU = 'mainarea-tabbar-context-menu';
+export const MAINAREA_TABBAR_CONTEXT_MENU = ['mainarea-tabbar-context-menu'];
```

Comment:
It would be better if you turn it into an array on the use-side

Reply:
Maybe we should declare a type alias `MenuPath` for `string[]`.
And then also change contextMenuRenderer to accept that type.

Reply:
ok